File: ios/engine/KMEI/KeymanEngine/Classes/LanguagePicker/KeyboardNameTableViewCell.m
//
//  KeyboardNameTableViewCell.m
//  Keyman Engine
//
//  Created by Patrick Weekes on 12-06-19.
//  Updated by Serkan Kurt on 30/08/2013.
//  Copyright (c) 2017 SIL International. All rights reserved.
//

#import "KeyboardNameTableViewCell.h"

static const NSInteger kKeyboardCellIsSelectedViewTag = 4576; // arbitrary number
//static const NSInteger kKeymanMinFileSizeFor3GWarning = 2000000; // in bytes

@implementation KeyboardNameTableViewCell

@synthesize indexPath = indexPath_;


- (id)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
	if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
		self.indentationLevel = 0;
	}
	return self;
}

- (void)setKeyboardState:(eKMKeyboardState)kbState selected:(BOOL)isSelected defaultAccessoryType:(UITableViewCellAccessoryType)accessoryType {
	switch (kbState) {
		case kKMKeyboardStateNeedsDownload:
		{
            self.accessoryView = nil;
			self.accessoryType = accessoryType;
			break;
		}
		case kKMKeyboardStateNeedsUpdate:
		{
            self.accessoryView = nil;
			self.accessoryType = accessoryType;
			break;
		}
		case kKMKeyboardStateUpToDate:
            self.accessoryView = nil;
			self.accessoryType = accessoryType;
			break;
		case kKMKeyboardStateDownloading:
		{
			//UIActivityIndicatorView *activityIndicator = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleGray];
			//self.accessoryView = activityIndicator;
			//self.accessoryType = UITableViewCellAccessoryNone;
			//[activityIndicator startAnimating];
            self.accessoryType = accessoryType;
			break;
		}
		case kKMKeyboardStateNone:
		default:
			self.accessoryView = nil;
			self.accessoryType = accessoryType;
			break;
	}
	
	UIView *isSelectedView = [self.contentView viewWithTag:kKeyboardCellIsSelectedViewTag];
	if (isSelected) {
		if (!isSelectedView) {
			isSelectedView = [[UIView alloc] initWithFrame:CGRectMake(0.0f, 0.0f, self.bounds.size.width, self.bounds.size.height)];
			isSelectedView.tag = kKeyboardCellIsSelectedViewTag;
			isSelectedView.backgroundColor = [UIColor greenColor];
			isSelectedView.autoresizingMask = UIViewAutoresizingFlexibleHeight|UIViewAutoresizingFlexibleWidth;
            [self.contentView addSubview:isSelectedView];
		}
	} else {
		[isSelectedView removeFromSuperview];
	}
}

@end

File: ios/engine/KMEI/KeymanEngine/Classes/LanguagePicker/KeyboardNameTableViewCell.swift
//
//  KeyboardNameTableViewCell.swift
//  KeymanEngine
//
//  Created by Gabriel Wong on 2017-09-12.
//  Copyright © 2017 SIL International. All rights reserved.
//

import UIKit

private let cellSelectedTag = 4576 // arbitrary number

class KeyboardNameTableViewCell: UITableViewCell, UIAlertViewDelegate {
  var indexPath: IndexPath?

  override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
    super.init(style: style, reuseIdentifier: reuseIdentifier)
    indentationLevel = 0
  }

  required init?(coder aDecoder: NSCoder) {
    fatalError("init(coder:) has not been implemented")
  }

  func setKeyboardState(_ kbState: KeyboardState, selected isSelected: Bool,
                        defaultAccessoryType accessoryType: UITableViewCellAccessoryType) {
    if kbState != .downloading {
      accessoryView = nil
    }
    self.accessoryType = accessoryType

    let isSelectedView = viewWithTag(cellSelectedTag)
    if isSelected {
      if isSelectedView == nil {
        let view = UIView(frame: CGRect(x: 0, y: 0, width: bounds.width, height: bounds.height))
        view.tag = cellSelectedTag
        view.backgroundColor = UIColor.green
        view.autoresizingMask = UIViewAutoresizing.flexibleWidth.union(.flexibleHeight)
        addSubview(view)
      }
    } else {
      isSelectedView?.removeFromSuperview()
    }
  }
}
